client-misk-spanner/src/test/kotlin/app/cash/backfila/client/misk/spanner/SpannerBackfillTest.kt

```diff
@@ -0,0 +1,161 @@
+package app.cash.backfila.client.misk.spanner
+
+import app.cash.backfila.client.BackfillConfig
+import app.cash.backfila.client.NoParameters
+import app.cash.backfila.embedded.Backfila
+import app.cash.backfila.embedded.createWetRun
+import com.google.cloud.spanner.DatabaseClient
+import com.google.cloud.spanner.DatabaseId
+import com.google.cloud.spanner.InstanceId
+import com.google.cloud.spanner.KeyRange
+import com.google.cloud.spanner.KeySet
+import com.google.cloud.spanner.Mutation
+import com.google.cloud.spanner.Mutation.WriteBuilder
+import com.google.cloud.spanner.Spanner
+import com.google.cloud.spanner.Statement
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.util.UUID
+import javax.inject.Inject
+
+@MiskTest(startService = true)
+class SpannerBackfillTest {
+  @Suppress("unused")
+  @MiskTestModule
+  val module = TestingModule()
+
+  @Inject
+  lateinit var backfila: Backfila
+  @Inject lateinit var spanner: Spanner
+  private lateinit var dbClient: DatabaseClient
+
+  @BeforeEach
+  fun setup() {
+    dbClient = createDbClient(
+      spanner, project = TestingModule.PROJECT_ID, instance = TestingModule.INSTANCE_ID, db = TestingModule.DB_ID,
+    )
+    val adminDbClient = spanner.databaseAdminClient
+    val adminClient = adminDbClient.getDatabase(TestingModule.INSTANCE_ID, TestingModule.DB_ID)
+    val columns = TrackData.COLUMNS.joinToString(",") {
+      "${it.name} STRING(128)"
+    }
+
+    val statement = "CREATE TABLE ${TrackData.TABLE_NAME}($columns) PRIMARY KEY (id)"
+    adminClient.updateDdl(listOf(statement), null).get()
+
+    dbClient.readWriteTransaction().run {
+      val mutation = Mutation.newInsertBuilder(TrackData.TABLE_NAME)
+        .setColumnTo(TrackData.Column.id.name, UUID.randomUUID().toString())
+        .setColumnTo(TrackData.Column.album_title.name, "album title")
+        .setColumnTo(TrackData.Column.album_token.name, "token")
+        .setColumnTo(TrackData.Column.track_title.name, "track title")
+        .setColumnTo(TrackData.Column.artist_name.name, "artist")
+        .build()
+      it.buffer(mutation)
+    }
+  }
+
+  @AfterEach
+  fun cleanup() {
```

Comment:
I wonder if I should port some of the grantly helpers into miskGcp...

Reply:
It would certainly reduce some of the need to reinvent the wheel in other projects.